File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
a = list(map(int, input().split()))
left, right = 0, 0
each = a[0]
ans = 0

while True:
    if each > m:
        if left == right:
            left += 1
            right += 1
            if right >= n:
                break
            each = a[right]
        else:
            each -= a[left]
            left += 1
    elif each < m:
        right += 1
        if right >= n:
            break
        each += a[right]
    else:
        ans += 1
        if left == right:
            left += 1
            right += 1
            if right >= n:
                break
            each = a[right]
        else:
            each -= a[left]
            left += 1

print(ans)

File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static int n, m, left, right, ans, sumValue;
    public static int[] a;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        a = new int[n];

        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(st.nextToken());
        }

        left = 0;
        right = 0;
        ans = 0;
        sumValue = a[0];

        while (true) {
            if (sumValue == m) {
                ans ++;
                right ++;
                if (right >= n) break;
                sumValue += a[right];
            } else if (sumValue > m) {
                sumValue -= a[left];
                left ++;
            } else {
                right ++;
                if (right >= n) break;
                sumValue += a[right];
            }
        }

        System.out.println(ans);
    }
}